llvm/test/Verifier/branch-weight.ll

```diff
@@ -36,4 +80,72 @@ define void @test(i1 %0) {
 
 !0 = !{!"function_entry_count", i32 1}
 
-; CHECK: expected either branch_weights or VP profile name
\ No newline at end of file
+; CHECK: expected either branch_weights or VP profile name
+
+;--- unknown-correct.ll
+declare void @to_invoke()
+declare i32 @__gxx_personality_v0(...)
+
+define void @invoker() personality ptr @__gxx_personality_v0 {
+  invoke void @to_invoke() to label %exit unwind label %lpad, !prof !0
+lpad:
+  %ll = landingpad {ptr, i32}
+  cleanup
+  ret void
+exit:
+  ret void
+}
+
+define i32 @test(i32 %a) {
+  %c = icmp eq i32 %a, 0
+  br i1 %c, label %yes, label %exit, !prof !0
+yes:
+  switch i32 %a, label %exit [ i32 1, label %case_b
+                               i32 2, label %case_c], !prof !0
+case_b:
+  br label %exit
+case_c:
+  br label %exit
+exit:
+  %r = select i1 %c, i32 1, i32 2, !prof !0
+  ret i32 %r
+}
+
+!0 = !{!"unknown"}
+
+;--- unknown-invalid.ll
+define void @test(i32 %a) {
+  %c = icmp eq i32 %a, 0
+  br i1 %c, label %yes, label %no, !prof !0
+yes:
+  ret void
+no:
+  ret void
+}
+
+!0 = !{!"unknown", i32 12, i32 67}
```

Comment:
What is the error message in this case?

Reply:
The `CHECK` one, i.e. "expected either branch_weights or VP profile name".

hmm.. maybe we want something like "`unknown` should have no operands"? My thinking was that if you had the operands then maybe you messed up the name.... idk. wdyt?